meinberlin/react/kiezradar/SearchProfiles.jsx

```diff
@@ -0,0 +1,294 @@
+import Spinner from '../contrib/Spinner'
+
+const titleText = django.gettext('Search Profiles')
+const descriptionText = django.gettext(
+  'In this area you manage your search profiles.'
+)
+const noSavedProfilesText = django.gettext('No saved search profiles')
+const findProjectsText = django.gettext('Find projects')
+const yourSavedProfilesText = django.gettext('Your saved search profiles')
+const renameText = django.gettext('Rename')
+const deleteText = django.gettext('Delete')
+const renameSearchProfileText = django.gettext('Rename search profile')
+const cancelText = django.gettext('Cancel')
+const saveText = django.gettext('Save')
+const savingText = django.gettext('Saving')
+const viewProjectsText = django.gettext('View projects')
+const errorText = django.gettext('Error')
+const errorSearchProfilesText = django.gettext(
+  'Failed to fetch search profiles'
+)
+const errorDeleteSearchProfilesText = django.gettext(
+  'Failed to delete search profile'
+)
+const errorUpdateSearchProfilesText = django.gettext(
+  'Failed to update search profile'
+)
+const alertHeadlineText = django.gettext(
+  'Search profile successfully deleted'
+)
+const alertText = django.gettext(
+  'Your changes have been deleted.'
+)
+
+export default function SearchProfiles (props) {
+  return (
+    <>
+      <h1>{titleText}</h1>
+      <p>{descriptionText}</p>
+      <SearchProfileList {...props} />
+    </>
+  )
+}
+
+function SearchProfileList ({ apiUrl, planListUrl }) {
+  const [alert, setAlert] = useState(false)
+  const [searchProfiles, setSearchProfiles] = useState([])
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState(null)
+
+  useEffect(() => {
+    const fetchSearchProfiles = async () => {
+      try {
+        setLoading(true)
+        setError(null)
+
+        const response = await fetch(apiUrl)
+
+        if (!response.ok) {
+          throw new Error(errorSearchProfilesText)
+        }
+
+        const data = await response.json()
+        setSearchProfiles(data)
+      } catch (err) {
+        setError(err.message)
+      } finally {
+        setLoading(false)
+      }
+    }
+    fetchSearchProfiles()
+  }, [])
+
+  if (loading) {
+    return <Spinner />
+  }
+
+  if (error) {
+    return (
+      <div className="search-profiles-list__error">
+        {errorText}: {error}
+      </div>
+    )
+  }
+
+  return (
+    <>
+      {alert && <Alert onClose={() => setAlert(false)} />}
+      {searchProfiles.length === 0
+        ? (
+          <>
+            <h2 className="search-profiles-list__title">{noSavedProfilesText}</h2>
+            <a href={planListUrl} className="button button--light">
+              <i className="fa-solid fa-magnifying-glass mr-1" />
+              {findProjectsText}
+            </a>
+          </>
+          )
+        : (
+          <>
+            <h2 className="search-profiles-list__title">
+              {yourSavedProfilesText} ({searchProfiles.length})
+            </h2>
```

Comment:
You can keep both the `h2`s in each of the cases (length === 0 // length !== 0) below the alert and just change the text inside the h2 based on the case and then skip the React.Fragments (<>)